backend/UserManagement.go

```diff
@@ -0,0 +1,234 @@
+func ExchangeToken(TempClientCode string) (ATokenResponse, error) {
+
+	cid, found := os.LookupEnv("LI_CLIENT_ID")
+	if !found {
+		return ATokenResponse{}, errors.New("Unable to get client id from env var")
+	}
+	csecret, found := os.LookupEnv("LI_CLIENT_SECRET")
+	if !found {
+		return ATokenResponse{}, errors.New("Unable to get client secret from env var")
+	}
+	ruri, found := os.LookupEnv("LI_REDIRECT_URI")
+	if !found {
+		return ATokenResponse{}, errors.New("Unable to get redirect uri from env var")
+	}
+	code := url.QueryEscape(TempClientCode)
+	ruri = url.QueryEscape(ruri)
+	content := fmt.Sprintf("grant_type=authorization_code&code=%s&redirect_uri=%s&"+
+		"client_id=%s&client_secret=%s", code, ruri, cid, csecret)
+
+	endpoint := "https://www.linkedin.com/oauth/v2/accessToken"
+	req, err := http.NewRequest("POST", endpoint, bytes.NewBuffer([]byte(content)))
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("Host", "www.linkedin.com")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return ATokenResponse{}, errors.New("Unable to call LI API")
+	}
+	defer resp.Body.Close()
+	// Fill the record with the data from the JSON response
+	var record ATokenResponse
+	if err := json.NewDecoder(resp.Body).Decode(&record); err != nil {
+		return ATokenResponse{}, errors.New("Unable to decode LI JSON")
+	}
+	if len(record.AToken) > 0 {
+		return record, nil
+	}
+	return ATokenResponse{}, errors.New("No access token returned from linkedIn API")
+}
+
+// GetLiProfile uses access token and REST call to get the user's linkedIn profile
+func GetLiProfile(AccessToken string) (LiProfile, error) {
+	// Fill the record with the data from the JSON
+	var record LiProfile
+	// QueryEscape escapes the parama
+	items := "(id,first-name,last-name,maiden-name,formatted-name,phonetic-first-name," +
+		"phonetic-last-name,formatted-phonetic-name,headline,location," +
+		"industry,current-share,num-connections,num-connections-capped," +
+		"summary,specialties,positions,picture-url,email-address)"
+
+	at := url.QueryEscape(AccessToken)
+	itm := url.QueryEscape(items)
+	url := fmt.Sprintf("%s/v1/people/~:%s?oauth2_access_token=%s&format=json", LIAPI, itm, at)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return LiProfile{}, errors.New("Unable to form HTTP request")
+	}
+	// For control over HTTP client headers,redirect policy, and other settings,
+	client := &http.Client{}
+	// returns an HTTP response
+	resp, err := client.Do(req)
+	if err != nil {
+		return LiProfile{}, errors.New("Unable to make REST call to get profile data")
+	}
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(&record); err != nil {
+		fmt.Println(err)
+	}
+	// temprary place holder
+	if err := json.Unmarshal([]byte(sampleProfile), &record); err != nil {
+		fmt.Println(err)
+	}
+	return record, nil
+}
+
+var sampleProfile = `
```

Comment:
Can we import the sample profile from notes.txt? Or consolidate the data locations?